src/ensembl/src/shared/checkbox/Checkbox.tsx

```diff
@@ -0,0 +1,64 @@
+import React, { useState } from 'react';
+import classNames from 'classnames';
+
+import defaultStyles from './Checkbox.scss';
+
+export enum CheckboxStatus {
+  CHECKED = 'checked',
+  DISABLED = 'disabled',
+  UNCHECKED = 'unchecked'
+}
+
+type Props = {
+  onClick?: () => void;
```

Comment:
1) Why would this function be optional? The component is useless without it.
2) Let's better call it `onChange` (to imitate the native input type checkbox more closely)

Reply:
> Why would this function be optional? The component is useless without it.
We don't have to worry about passing an onChange function if it is disabled?

> Let's better call it onChange (to imitate the native input type checkbox more closely)
Cool, will update it.

Reply:
> We don't have to worry about passing an onChange function if it is disabled?

Usually the value of the `disabled` prop will be computed at render time and passed to the component along with other properties that it needs:

```
<Checkbox
  checked={props.something}
  onChange={props.someFuncton}
  disabled={someCondition}
/>
```
which, typically, means that `onChange` is going to be passed to the component anyway, so this shouldn't be a problem. Curious whether you've noticed other patterns.

_(The `Input` component currently has `onChange` as a mandatory prop)_

Reply:
> which, typically, means that onChange is going to be passed to the component anyway

Yes, that's right. It's always going to be passed regardless of the state. I'll set it to be mandatory.